sdk/core/azure-core/test/ut/curl_session.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <string>
+//#define BUILD_CURL_HTTP_TRANSPORT_ADAPTER
+#include <azure/core/http/curl/curl.hpp>
+
+namespace Azure { namespace Core { namespace Test {
+
+  class CurlSession : public ::testing::Test {
+  };
+
+  /**
+   * @brief mock the network connection
+   *
+   */
+  class CurlNetworkConnectionMock : public Azure::Core::Http::CurlNetworkConnection {
```

Comment:
I suggest we standardize on mock being a prefix.  It reads better when its written "Mock of X" vs "X Mock"

```suggestion
  class MockCurlNetworkConnection : public Azure::Core::Http::CurlNetworkConnection {
```